sdk/purview/purview-administration-rest/review/purview-administration.api.md

```diff
@@ -0,0 +1,1314 @@
+    "x-ms-error-code"?: string;
+}
+
+// @public
+interface MetadataRolesListdefaultResponse extends HttpResponse {
+    // (undocumented)
+    body: ErrorResponseModel_2;
+    // (undocumented)
+    headers: RawHttpHeaders & MetadataRolesListdefaultHeaders;
+    // (undocumented)
+    status: "500";
+}
+
+// @public (undocumented)
+type MetadataRolesListParameters = RequestParameters;
+
+declare namespace Models {
+    export {
+        Account,
+        Identity,
+        AccountProperties,
+        CloudConnectors,
+        AccountEndpoints,
+        AccountPropertiesEndpoints,
+        ManagedResources,
+        AccountPropertiesManagedResources,
+        PrivateEndpointConnection,
+        PrivateEndpointConnectionProperties,
+        PrivateEndpoint,
+        PrivateLinkServiceConnectionState,
+        AccountSku,
+        SystemData,
+        AccountSystemData,
+        ErrorResponseModel,
+        ErrorModel,
+        ErrorResponseModelError,
+        DataPlaneAccountUpdateParameters,
+        AccessKeys,
+        AccessKeyOptions,
+        Collection,
+        CollectionReference,
+        CollectionSystemData,
+        CollectionList,
+        CollectionNameResponseList,
+        CollectionNameResponse,
+        CollectionPathResponse,
+        ResourceSetRuleConfig,
+        AdvancedResourceSet,
+        PathPatternExtractorConfig,
+        Filter,
+        ComplexReplacerConfig,
+        NormalizationRule,
+        FastRegex,
+        RegexReplacer,
+        ScopedRule,
+        Rule,
+        ResourceSetRuleConfigList
+    }
+}
+
+declare namespace Models_2 {
+    export {
+        MetadataRoleList,
+        MetadataRole,
+        MetadataRoleProperties,
+        AttributeMatcher,
+        ErrorResponseModel_2 as ErrorResponseModel,
+        ErrorModel_2 as ErrorModel,
+        MetadataPolicyList,
+        MetadataPolicy,
+        MetadataPolicyProperties,
+        DecisionRule,
+        AttributeRule,
+        CollectionReference_2 as CollectionReference
+    }
+}
+
+// @public (undocumented)
+interface NormalizationRule {
+    // (undocumented)
+    description?: string;
+    // (undocumented)
+    disabled?: boolean;
+    // (undocumented)
+    dynamicReplacement?: boolean;
+    // (undocumented)
+    entityTypes?: Array<string>;
+    // (undocumented)
+    lastUpdatedTimestamp?: number;
+    // (undocumented)
+    name?: string;
+    // (undocumented)
+    regex?: FastRegex;
+    // (undocumented)
+    replaceWith?: string;
+    // (undocumented)
+    version?: number;
+}
+
+// @public
+function paginate<TResponse extends PathUncheckedResponse>(client: Client, initialResponse: TResponse, options?: PagingOptions<TResponse>): PagedAsyncIterableIterator<PaginateReturn<TResponse>>;
+
+// @public
+function paginate_2<TResponse extends PathUncheckedResponse>(client: Client, initialResponse: TResponse, options?: PagingOptions_2<TResponse>): PagedAsyncIterableIterator<PaginateReturn_2<TResponse>>;
```

Comment:
no need to generate two `paginate`s.

Reply:
They are generated slightly differently for each of the clients. Both are grouped in its own sub-module. `PurviewAccount.Pagination.paginate()` and `PurviewMetadataPolicies.Pagination.paginate()`

Reply:
I do not see a difference in their signatures, is there? also, is there any differences in the implementation? If not, did you consider generating just one under a common module?

Reply:
Yeah, the difference is in the implementation. Metadata service may return the page in the `value` or `values` property in the body while Account service only in the `value` property. Having different possibilities for the page requires some extra code to be generated to extract the page.

Reply:
The common module is a good idea and something to consider when we work on the actual generation of multi-client packages. If LRO or Pagination has the same implementations we may put them there, the same thing could be done with common models.